doc/man3/SSL_CTX_set_mode.pod

```diff
@@ -51,16 +51,20 @@ non-blocking write().
 
 =item SSL_MODE_AUTO_RETRY
 
-Never bother the application with retries if the transport is blocking.
-If a renegotiation take place during normal operation, a
-L<SSL_read_ex(3)>, L<SSL_read(3)>, L<SSL_write_ex(3)> or L<SSL_write(3)> would
-return with a failure and indicate the need to retry with SSL_ERROR_WANT_READ.
+During normal operations, non-applications data records might need to be sent or
+received that the application is not aware of.
+If a non-application data record was processed, a
+L<SSL_read_ex(3)>, L<SSL_read(3)>, L<SSL_write_ex(3)> or L<SSL_write(3)> can
+return with a failure and indicate the need to retry with B<SSL_ERROR_WANT_READ>.
+If such a non-application data record was processed, the flag
+B<SSL_MODE_AUTO_RETRY> causes it to try to process the next record instead of
+returning.
 In a non-blocking environment applications must be prepared to handle
 incomplete read/write operations.
 In a blocking environment, applications are not always prepared to
-deal with read/write operations returning without success report. The
-flag SSL_MODE_AUTO_RETRY will cause read/write operations to only
-return after the handshake and successful completion.
+deal with read/write operations returning without success report,
+and setting the B<SSL_MODE_AUTO_RETRY> flag will cause them to only
+return after success or failure.
```

Comment:
This is a bit confusing wording. I mean it first suggests that application can't handle operations without success report, and then says that flags will make operations to only return after success or failure. Maybe it would be more appropriate to put is as "applications can't deal with *intermediate* reports such as retry requests.". I also find "them" ambiguous. It's not clear if it refers to applications or operations. So "them, the operations" perhaps?